### PR TITLE
Added the auto-generated settings documentation deployment

### LINKED ISSUE #2858: Automatically update default settings documentation
Do the same with https://github.com/OpenRA/OpenRA/wiki/Settings as #2808 does with https://github.com/OpenRA/OpenRA/wiki/Traits
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/136903-automatically-update-default-settings-documentation?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Mailaender**
Followup of https://github.com/OpenRA/OpenRA/pull/11571 to finally fix https://github.com/OpenRA/OpenRA/issues/2858.

### CODE REVIEW

File: packaging/windows/OpenRA.nsi [hunk 1]
```diff
@@ -201,6 +202,7 @@ Function ${UN}Clean
 	Delete $INSTDIR\CHANGELOG.html
 	Delete $INSTDIR\CONTRIBUTING.html
 	Delete $INSTDIR\DOCUMENTATION.html
+	Delete $INSTDIR\Settings.html
```
Comment (obrakmann) [inline packaging/windows/OpenRA.nsi]: While you're at it, could you add the Lua-API.html file here as well? It seems to be missing.

File: packaging/package-all.sh [hunk 2]
```diff
@@ -29,12 +29,13 @@ markdown README.md > README.html
...
 markdown DOCUMENTATION.md > DOCUMENTATION.html
 markdown Lua-API.md > Lua-API.html
+markdown Settings.md > Settings.html
 
 # List of files that are packaged on all platforms
 FILES=('OpenRA.Game.exe' 'OpenRA.Game.exe.config' 'OpenRA.Utility.exe' 'OpenRA.Server.exe' 
 'OpenRA.Platforms.Default.dll' \
 'lua' 'glsl' 'mods/common' 'mods/ra' 'mods/cnc' 'mods/d2k' 'mods/modchooser' \
-'AUTHORS' 'COPYING' 'README.html' 'CONTRIBUTING.html' 'DOCUMENTATION.html' 'CHANGELOG.html' \
+'AUTHORS' 'COPYING' 'README.html' 'CONTRIBUTING.html' 'DOCUMENTATION.html' 'CHANGELOG.html' ' Settings.html' \
```
Comment (obrakmann) [inline packaging/package-all.sh]: The Lua-API.html file is missing here as well.

**Comment by obrakmann** [comment]
Looks otherwise ok, :+1:

**Comment by Mailaender** [comment]
Updated.

File: packaging/windows/OpenRA.nsi [hunk 3]
```diff
@@ -96,6 +96,8 @@ Section "Game" GAME
 	File "${SRCDIR}\CHANGELOG.html"
 	File "${SRCDIR}\CONTRIBUTING.html"
 	File "${SRCDIR}\DOCUMENTATION.html"
+	File "${SRCDIR}\Lua-API.html"
```
Comment (pchote) [inline packaging/windows/OpenRA.nsi]: Please rename these to match the all-caps names of the other html files.

File: Makefile [hunk 4]
```diff
@@ -370,6 +370,7 @@ version: mods/ra/mod.yaml mods/cnc/mod.yaml mods/d2k/mod.yaml mods/ts/mod.yaml m
 docs: utility mods version
 	@mono --debug OpenRA.Utility.exe all --docs > DOCUMENTATION.md
```
Comment (pchote) [inline Makefile]: If we're going to be shipping multiple API files with the installer then this should be renamed to TRAIT-API.md.

**Comment by pchote** [comment]
I personally think that the hassle of maintaining the continuously-breaking md -> html infrastructure far outweighs the benefits of shipping trait/lua documentation files that I doubt anybody has ever read.  I'd much prefer we remove the misnamed DOCUMENTATION.html from the installers instead of shipping even more files that people won't read.

File: Makefile (hunk 4 above)
Comment (Mailaender) [inline Makefile]: It is not really an application programming interface. I will rename it to match the wiki name instead.

**Comment by Mailaender** [comment]
Your request of removing the MarkDown to HTML "infrastructure" due to volatility comes many months too late. All problems are solved and they were rather introduced by Windows package management being still in its infancy. Removing all offline documentation is definitely not going to happen.

File: Makefile [hunk 5]
```diff
@@ -368,8 +368,9 @@ version: mods/ra/mod.yaml mods/cnc/mod.yaml mods/d2k/mod.yaml mods/ts/mod.yaml m
 	done
 
 docs: utility mods version
-	@mono --debug OpenRA.Utility.exe all --docs > DOCUMENTATION.md
-	@mono --debug OpenRA.Utility.exe all --lua-docs > Lua-API.md
+	@mono --debug OpenRA.Utility.exe all --docs > TRAITS.md
+	@mono --debug OpenRA.Utility.exe all --lua-docs > LUA-API.md
+	@mono --debug OpenRA.Utility.exe all --settings-docs > Settings.md
```
Comment (pchote) [inline Makefile]: This is still inconsistent with the rest of the filenames.

Comment (pchote) [inline Makefile]: Having all three pages in regular case would also be fine by me.  The important point is that they agree.

Comment (Mailaender) [inline Makefile]: Renamed.

File: packaging/windows/OpenRA.nsi [hunk 6]
```diff
@@ -200,7 +202,9 @@ Function ${UN}Clean
 	Delete $INSTDIR\README.html
 	Delete $INSTDIR\CHANGELOG.html
 	Delete $INSTDIR\CONTRIBUTING.html
-	Delete $INSTDIR\DOCUMENTATION.html
+	Delete $INSTDIR\TRAITS.html
+	Delete $INSTDIR\Lua-API.html
```
Comment (pchote) [inline packaging/windows/OpenRA.nsi]: Filename case here needs fixing.

Comment (Mailaender) [inline packaging/windows/OpenRA.nsi]: Right. Fixed.

**Comment by pchote** [comment]
The output this generates includes a section on the launch parameters, which have nothing to do with the settings infrastructure.  This is quite confusing.

**Comment by Mailaender** [comment]
I disagree and that request also has nothing to do with this pull request. You should have commented on https://github.com/OpenRA/OpenRA/pull/11571 and as it already had reached consensus you can submit your own patch and find voters for it.

**Comment by pchote** [comment]
This PR is about shipping a file called SETTINGS.html with the implication that it is documenting the game's settings.yaml.  The  Launch block is right in the middle of this file (between the graphics and server blocks), with no indication that this has nothing to do with settings.yaml.  The first sentence of the page claims "This documentation is aimed at server administrators.", but those arguments aren't used for servers either.

We should not be shipping incorrect documentation, so we should not be merging this without clarification to either the purpose of the documentation, or its content. :-1:

Also shame on you for calling review feedback invalid because *I didn't catch it early enough*.  I had made it clear well in advance that I was going to be busy with work for all of July and most of August, and that I wouldn't be available for seriously reviewing PRs.  There wasn't any indication from the notification chatter that those docs included the Launch block.

**Comment by Mailaender** [comment]
As I already said: be my guest. File a pull request to modify the settings extraction output yourself. You are not the only one busy with work.

**Comment by pchote** [comment]
Moving to the future milestone as part of PR queue cleanup.

**Comment by Mailaender** [comment]
>  I'd much prefer we remove the misnamed DOCUMENTATION.html from the installers instead of shipping even more files that people won't read.

I did just that now after https://github.com/OpenRA/OpenRA/pull/13215/ has been merged. Also fixed the other minor requested changes.

Preview: https://github.com/OpenRA/OpenRA/wiki/Settings

**Comment by pchote** [comment]
I'd like to apologise for my comment above.  While I stand by the points I raised, the tone I used was uncalled for and unacceptable.  I was going through a stressful time then, and let frustration get the better of me.

**Comment by Mailaender** [comment]
Nevermind. I had exactly the same feeling. Not sure how it escalated that quickly over such minor issues. Also added some field descriptors that we described manually before  https://github.com/OpenRA/OpenRA/wiki/Settings/_compare/7efefffd47c07b53dc384571edc9f4f686daecac...1034cb2b44b540429e6500407ad919a5ddbc05ec

**Comment by pchote** [comment]
> This documentation is aimed at server administrators.

This is no longer really true, the server will ignore most of these settings and we have https://github.com/OpenRA/OpenRA/blob/bleed/launch-dedicated.sh to document the settings that it does expect.

Can you please update the header to reflect that this is documenting settings.yaml for the game / players?

**Comment by Mailaender** [comment]
Yes, done.

**Review by pchote — APPROVED** [review]
It would be good if you could squash down some of the commits, but I won't block merging over that.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/45ae80d3c029f87c33bda5c5b0d1631e6c67740f...0a50c65d209005d3d1fabc8d5a8be1efd12b1e0f

**Pull request merged**